### PR TITLE
adding webhook templates

### PR DESCRIPTION
**Pull request by mattchenderson**
Targeting dev as this can't go directly onto 2.x. Let me know if/how we want to rebase onto 2.x instead.

### CODE REVIEW

File: Functions.Templates/Templates/OutlookMessageWebhookDeleter-CSharp/run.csx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using Microsoft.Graph;
+using System.Net;
+
+public static async Task Run(Message msg, string[] existingSubscriptions, ICollector<string> subscriptionsToDelete, TraceWriter log)  
+{
+    log.Info("C# HTTP trigger function processed a request.");
+	foreach (var subscription in existingSubscriptions)
+	{
+		log.Info($"Deleting subscription {subscription}");
+        subscriptionsToDelete.Add(subscription);
```
Comment (soninaren) [inline Functions.Templates/Templates/OutlookMessageWebhookDeleter-CSharp/run.csx]: Nit: Indentation

File: Functions.Templates/Templates/OutlookMessageWebhookHandler-CSharp/metadata.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+﻿{
+  "defaultFunctionName": "OultookMessageWebhookHandlerCSharp",
+  "description": "$OultookMessageWebhookHandlerCSharp_description",
+  "name": "Outlook webhook handler",
+  "trigger": "HttpTrigger",
```
Comment (soninaren) [inline Functions.Templates/Templates/OutlookMessageWebhookHandler-CSharp/metadata.json]: unused property

File: Functions.Templates/Templates/OutlookMessageWebhookHandler-CSharp/run.csx [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+using System.Net;
+
+public static async Task Run(Message msg, TraceWriter log)  
+{
+	log.Info("Microsoft Graph webhook trigger function processed a request.");
+
+    // Testable by sending oneself an email with the Subject "Azure Functions" and some text body
+    if (msg.Subject.Contains("Azure Functions") && msg.From.Equals(msg.Sender)) {
+        log.Info($"Processed email: {msg.BodyPreview}");
+    }
```
Comment (soninaren) [inline Functions.Templates/Templates/OutlookMessageWebhookHandler-CSharp/run.csx]: Nit: Indentation

File: Functions.Templates/Templates/OutlookMessageWebhookRefresher-CSharp/run.csx [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+public static void Run(TimerInfo myTimer, string[] existingSubscriptions, ICollector<string> subscriptionsToRefresh, TraceWriter log)
+{
+    // This template uses application permissions and requires consent from an Azure Active Directory admin.
+    // See https://go.microsoft.com/fwlink/?linkid=858780
+    log.Info($"C# Timer trigger function executed at: {DateTime.Now}");
+	foreach (var subscription in existingSubscriptions)
+	{
+		log.Info($"Refreshing subscription {subscription}");
+        subscriptionsToRefresh.Add(subscription);
```
Comment (soninaren) [inline Functions.Templates/Templates/OutlookMessageWebhookRefresher-CSharp/run.csx]: Nit: Indentation

File: Functions.Templates/Templates/OutlookMessageWebhookRefresher-CSharp/function.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+﻿{
+  "bindings": [
+    {
+      "name": "myTimer",
+      "type": "timerTrigger",
+      "direction": "in",
+      "schedule": "0 * * */2 * *"
```
Comment (soninaren) [inline Functions.Templates/Templates/OutlookMessageWebhookRefresher-CSharp/function.json]: may want to provide user the option to change this at the time of template creation

File: Functions.Templates/Templates/OutlookMessageWebhookRefresher-CSharp/run.csx (hunk 4 above)
Comment (mattchenderson) [reply Functions.Templates/Templates/OutlookMessageWebhookRefresher-CSharp/run.csx]: Not sure what's meant here. Will review in person.

File: Functions.Templates/Templates/OutlookMessageWebhookHandler-CSharp/metadata.json (hunk 2 above)
Comment (mattchenderson) [reply Functions.Templates/Templates/OutlookMessageWebhookHandler-CSharp/metadata.json]: removed in all of these templates

File: Functions.Templates/Templates/OutlookMessageWebhookDeleter-CSharp/function.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    {
+      "type": "graphWebhookSubscription",
+      "name": "subscriptionsToDelete",
+      "direction": "out",
+      "action": "delete",
+      "identity": "userFromRequest"
+    },
+    {
+      "type": "http",
+      "name": "res",
```
[the remark this answers is not in the document]
Comment (mattchenderson) [reply Functions.Templates/Templates/OutlookMessageWebhookDeleter-CSharp/function.json]: changed in all of these templates

**Comment by mattchenderson** [comment]
About to make some changes to align with documentation.

**Pull request merged**